src/main/java/com/cursor/restpractic/web/TeacherController.java

```diff
@@ -0,0 +1,42 @@
+package com.cursor.restpractic.web;
+
+import com.cursor.restpractic.dto.TeacherDto;
+import com.cursor.restpractic.model.GroupStudents;
+import com.cursor.restpractic.service.TeacherService;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+import static org.springframework.http.HttpStatus.CREATED;
+
+@RestController
+@RequestMapping(value = "/api/teachers")
+@RequiredArgsConstructor
+public class TeacherController {
+
+    private final TeacherService teacherService;
+
+    @PostMapping(value = "/create")
+    public ResponseEntity<TeacherDto> create(@Valid @RequestBody TeacherDto teacherDto) {
+        return new ResponseEntity<TeacherDto>(teacherService.create(teacherDto), CREATED);
+    }
+
+    @PutMapping(value = "/add/{teacherId}/{groupId}")
```

Comment:
verbs in url's is not a good practice, it's not a REST approach, u should use http method for action, not declare it on url's